storage.go

```diff
@@ -79,6 +102,48 @@ func NewDiskImageStorageDeviceAttachment(diskPath string, readOnly bool) (*DiskI
 	return attachment, nil
 }
 
+// NewDiskImageStorageDeviceAttachmentWithCacheAndSync initialize the attachment from a local file path.
+// Returns error is not nil, assigned with the error if the initialization failed.
+//
+// - diskPath is local file URL to the disk image in RAW format.
+// - readOnly if YES, the device attachment is read-only, otherwise the device can write data to the disk image.
+// - cachingMode is one of the available DiskImageCachingMode options.
+// - syncMode is to define how the disk image synchronizes with the underlying storage when the guest operating system flushes data, described by one of the available DiskImageSynchronizationMode modes.
+//
+// This is only supported on macOS 12 and newer, error will
+// be returned on older versions.
+func NewDiskImageStorageDeviceAttachmentWithCacheAndSync(diskPath string, readOnly bool, cachingMode DiskImageCachingMode, syncMode DiskImageSynchronizationMode) (*DiskImageStorageDeviceAttachment, error) {
```

Comment:
Could you add tests here?

This is to check availability.
https://github.com/Code-Hex/vz/blob/23eeededdc039cdcbab348836d84432891aea7ca/osversion_test.go#L109

This one is for checking actually working or not. (It's OK only check exists disk path)
https://github.com/Code-Hex/vz/blob/main/storage_test.go

Reply:
Done the first test, about second test we don't have disk path from configuration i believe.
I can add a test to create disk (with cache and sync mode) and start vm and see if its running. Is that okay ?

Reply:
@balajiv113 Sure! Thanks!